spec/booking_manager_spec.rb

```diff
@@ -0,0 +1,333 @@
+require_relative 'spec_helper'
+require 'date'
+require 'pry'
+
+describe "Booking Manager class" do
+
+  describe 'Hotel Manager instantiation' do
+    before do
+      @hotel = Hotel::Booking_Manager.new
+    end
+
+    it 'has an array of all rooms in hotel' do
+      expect(@hotel.rooms).must_be_kind_of Array
+    end
+
+    it 'has 20 rooms in the hotel' do
+      expect(@hotel.rooms.length).must_equal 20
+    end
+
+    it 'each index of the array is an instance of class' do
+      expect(@hotel.rooms[0]).must_be_kind_of Hotel::Room
+    end
+  end
+
+  describe 'get_room method' do
+    before do
+      @hotel = Hotel::Booking_Manager.new
+    end
+
+    it 'returns an Array of rooms in hotel' do
+      expect(@hotel.get_rooms).must_be_kind_of Array
+    end
+
+    it 'returns an Array of correct length' do
+        expect(@hotel.get_rooms.length).must_equal 20
+    end
+  end
+
+  describe 'reserve_room method' do
+    before do
+      @hotel = Hotel::Booking_Manager.new
+      @date1 = Date.new(2018, 6, 6)
+      @date2 = Date.new(2018, 6, 9)
+      @all_rooms_available = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20]
+      @input = { name: "Dr. Frankenstein",
+        room_number: 6,
+        check_in_date: @date1,
+        check_out_date: @date2
+      }
+    end
+
+    it 'adds a reservation to the array of all reservations for a room' do
+      @hotel.reserve_room(@input)
+      expect(@hotel.rooms[5].reservations[0].name).must_equal "Dr. Frankenstein"
```

Comment:
As much as I love Dr. Frankenstein gettin' a reservation at the ole Hotel Ada, is there a requirement in the project to have a name associated?